resources/type-system/inference.md

```diff
@@ -324,7 +329,8 @@ schema.
     `e`, using the local type inference algorithm described below with typing
     context `K`, and update `T` to be `UP(flatten(S), T)` if the enclosing
     function is `async`, or `UP(S, T)` otherwise.
-  - For each `return;` statement in the block, update `T` to be `UP(Null, T)`.
+  - If the enclosing function is not marked `sync*` or `async*`: For each
```

Comment:
The "for each" seems redundant. How about "if the body contains any `return;` statement ..."?

Reply:
All four items in this list use a similar quantifier (actually, they all use 'for each'). How would it be redundant? We're running an algorithm, and at this point it needs to iterate over the function body to find all occurrences of `return ...`. Do you think it would improve the overall text to use a natural language version of recursion, rather than the iteration-ish "for each"?